code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/DeploymentUnitInstance.java

```diff
@@ -42,21 +47,60 @@ public boolean evaluate(ServiceExposeMap obj) {
 
     public abstract void stop();
 
+    @SuppressWarnings("unchecked")
     protected DeploymentUnitInstance(DeploymentServiceContext context, String uuid, Service service,
             String launchConfigName) {
         this.context = context;
         this.uuid = uuid;
         this.launchConfigName = launchConfigName;
         this.service = service;
+        Object policy = ServiceDiscoveryUtil.getLaunchConfigObject(service, launchConfigName,
+                DockerInstanceConstants.FIELD_RESTART_POLICY);
+
+        if (policy != null && ((Map<String, String>) policy).get("name").equalsIgnoreCase(RestartPolicy.RESTART_NEVER)) {
```

Comment:
Might want to switch it around to:
RestartPolicy.RESTART_NEVER.equalsIgnoreCase(((Map<String, String>) policy).get("name")) just in case ((Map<String, String>) policy).get("name") is null